CMakeLists.txt

```diff
@@ -104,18 +104,40 @@ endif()
 ocv_cmake_hook(CMAKE_INIT)
 
 # must go before the project()/enable_language() commands
-ocv_update(CMAKE_CONFIGURATION_TYPES "Debug;Release" CACHE STRING "Configs" FORCE)
-if(NOT DEFINED CMAKE_BUILD_TYPE
-    AND NOT OPENCV_SKIP_DEFAULT_BUILD_TYPE
-)
-  message(STATUS "'Release' build type is used by default. Use CMAKE_BUILD_TYPE to specify build type (Release or Debug)")
-  set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build")
+if(NOT DEFINED CMAKE_BUILD_TYPE AND NOT OPENCV_SKIP_DEFAULT_BUILD_TYPE)
+  # detects if the generator is a known multi-configuration generator
+  if(CMAKE_GENERATOR MATCHES "Visual Studio|Xcode|Ninja Multi-Config")
+    message(STATUS "Using multi-configuration generator: ${CMAKE_GENERATOR}.")
+    message(STATUS "'Debug' configuration will be used by default. Use the IDE to switch between configurations (Debug, Release, etc.).")
+    # sets the default configurations for multi-configuration generators
+    ocv_update(CMAKE_CONFIGURATION_TYPES "Debug;Release" CACHE STRING "Configs" FORCE)
+  
+  elseif(CMAKE_GENERATOR MATCHES "Unix Makefiles|Ninja|Watcom WMake")
+    # for single-configuration generators
+    if(CMAKE_GENERATOR MATCHES "Unix Makefiles")
+      message(STATUS "Unix Makefiles = Generates standard UNIX makefiles.")
+    elseif(CMAKE_GENERATOR MATCHES "Ninja")
+      message(STATUS "Ninja = Generates build.ninja files.")
+    elseif(CMAKE_GENERATOR MATCHES "Watcom WMake")
+      message(STATUS "Watcom WMake = Generates Watcom WMake makefiles.")
+    endif()
```

Comment:
It's part of `cmake --help` message. Why do we need it here?

Reply:
Thank you for the feedback, asmorkalov.

The goal of this change is to provide immediate, context-aware messages to developers during the configuration process, as suggested by vpisarev. The current default message:
```
'Release' build type is used by default. Use CMAKE_BUILD_TYPE to specify build type (Release or Debug)
```
is accurate for Unix Makefiles but not for Visual Studio or Xcode, where 'Debug' is the default. To enhance clarity:

- For multi-configuration generators (Visual Studio, Xcode, Ninja Multi-Config):
  - Display a message indicating that 'Debug' is the default and that users can switch configurations within the IDE.
  
- For single-configuration generators (Unix Makefiles, Ninja, Watcom WMake):
  - Display a message indicating that 'Release' is the default and that `CMAKE_BUILD_TYPE` can be specified to change this.

These changes aim to reduce confusion and improve the user experience by making the default build types clear during configuration, without needing to refer to additional documentation.

If you have any suggestions for changes or improvements, I am ready to incorporate them.

Thank you for considering this.